backend/src/controllers/notes.ts

```diff
@@ -140,10 +170,105 @@ export const deleteNote: RequestHandler = async (req, res, next) => {
       throw createHttpError(401, "You cannot access this note");
     }
 
-    await note.remove();
+    note.deletedAt = new Date();
+
+    await note.save();
```

Comment:
why do we do `note.save()` rather than `note.remove()`?

Reply:
Cuz I change the delete logic into trash system.  I want to keep them in trash instead of delete